generators/csharp/codegen/src/ast/XmlDocBlock.ts

```diff
@@ -0,0 +1,110 @@
+import { AstNode, Writer } from "../csharp";
+import { XmlDocWriter } from "./core/XmlDocWriter";
+
+export declare namespace XmlDocBlock {
+    type Like = XmlDocBlock | XmlDocBlock.Arg;
+    type Arg =
+        | {
+              summary?: XmlDocNode;
+              codeExample?: XmlDocNode;
+              exceptions?: Map<string | AstNode, XmlDocNode>;
```

Comment:
nit: Should we name this type too? That way it can be individually referred to.

Reply:
Done